app/src/main/java/com/micahThor/taskmaster/AddTaskActivity.java

```diff
@@ -126,4 +178,48 @@ public void onFailure(@Nonnull ApolloException e) {
             Log.e("micah", e.toString());
         }
     };
+
+    public void uploadWithTransferUtility() {
+        TransferUtility transferUtility =
+                TransferUtility.builder()
+                        .context(getApplicationContext())
+                        .awsConfiguration(AWSMobileClient.getInstance().getConfiguration())
+                        .s3Client(new AmazonS3Client(AWSMobileClient.getInstance()))
+                        .build();
+
+
+//        Log.i("PATHNAME---------------", imageUri);
+//        File file = new File(imageUri);
```

Comment:
Pretty sure to make a new file you'll need the path, not the uri. There's a way to get the path from a uri;

    public String getPath(Uri uri)
    {
        String[] projection = { MediaStore.Images.Media.DATA };
        Cursor cursor = getContentResolver().query(uri, projection, null, null, null);
        if (cursor == null) return null;
        int column_index =             cursor.getColumnIndexOrThrow(MediaStore.Images.Media.DATA);
        cursor.moveToFirst();
        String s=cursor.getString(column_index);
        cursor.close();
        return s;
    }

Something similar an be found at the bottom of; https://aws-amplify.github.io/docs/sdk/android/storage
Also from Michelle's lecture yesterday.